MediaProcessor/src/AudioProcessor.cpp

```diff
@@ -15,19 +15,40 @@ AudioProcessor::AudioProcessor(const std::string &inputVideoPath,
                                const std::string &outputAudioPath)
     : m_inputVideoPath(inputVideoPath),
       m_outputAudioPath(outputAudioPath),
-      m_numChunks(DEFAULT_NUM_CHUNKS),
+      m_numChunks(determineNumChunks()),
       m_overlapDuration(DEFAULT_OVERLAP_DURATION) {
     m_outputDir = std::filesystem::path(outputAudioPath).parent_path().string();
     m_chunksDir = m_outputDir + "/chunks";
     m_processedChunksDir = m_outputDir + "/processed_chunks";
 
+    std::cout << "INFO: using " << std::to_string(m_numChunks) << " Threads." << std::endl;
+
     /*
      * TODO: 
      * * numChunks (numThreads) should be configurable, as well as the overlap (which seems much less critical?)
      * * Need an ffmpeg utils to establish an API for extaction and probing
      */
 }
 
+int AudioProcessor::determineNumChunks() const {
+    /*
+     * If the value is not well-defined or not computable, hardware_concurrency() may return 0.
+     * In that case, we'll fall back to DEFAULT_NUM_CHUNKS.
+     * We subtract 2 from the available hardware threads as a safety margin to avoid overloading the system.
+     */
+    ConfigManager &configManager = ConfigManager::getInstance();
+    int configNumChunks = configManager.getNumChunks();
+    int hardwareNumChunks = std::thread::hardware_concurrency();
+
+    if (configNumChunks >= 1 && configNumChunks <= hardwareNumChunks) {
```

Comment:
Please add an `isWithinRange` to `Utils`, so that this reads `if(isWithinRange)` which is much easier to reason about